add: search/transactions (#2986)
dfinity/portal: docs/developer-docs/defi/rosetta/icrc_rosetta/data_api/transactions.mdx

<MarkdownChipRow labels={["Intermediate", "Tutorial" ]} />

## Overview 

> There exist two different entpoints that allow for querying of transaction. 

Reviewer: ```suggestion
There are two endpoints that allow for querying transactions. 
```

The `block_transaction` endpoint allows you to fetch a transaction at a certain block height. It is the implementation of the [/block/transaction endpoint](https://docs.cdp.coinbase.com/rosetta/reference/blocktransaction) of the Rosetta API standard. 

> The `search_transactions` endpoint allows you to query for transactions given a set of parameters. It is the implementation of the [/search/transactions endpoint](https://docs.cdp.coinbase.com/rosetta/reference/searchtransactions/)

Reviewer: ```suggestion
The `search_transactions` endpoint allows you to query for transactions given a set of parameters. It is the implementation of the [/search/transactions endpoint](https://docs.cdp.coinbase.com/rosetta/reference/searchtransactions/).
```

        }
    }
}
```

## SearchTransactions endpoint

> There exist multiple parameters that can be used to query for transactions. See the [full specification](https://docs.cdp.coinbase.com/rosetta/reference/searchtransactions/) to see what query parameters can be used. 
> For example if you want to fetch all transactions in a given block range of maximal 10000 blocks where a specific account was involved in you can make a request to the `search/transactions` endpoint like the following:

Reviewer: ```suggestion
There are several parameters that can be used to query transactions. See the [full specification](https://docs.cdp.coinbase.com/rosetta/reference/searchtransactions/) to see what query parameters can be used. 
For example, if you want to fetch all transactions in a given block (range of maximal 10000 blocks) where a specific account was involved, you can make a request to the `search/transactions` endpoint:
```

```json
curl --location '0.0.0.0:8081/search/transactions'  --header 'Content-Type: application/json' --data '{
    "network_identifier": {
        "blockchain": "Internet Computer",
        "network":"2ouva-viaaa-aaaaq-aaamq-cai"
    },
     "account_identifier": {
        "address": "h2upz-svoky-hpfgf-lobao-afowo-h6xs6-4jhj5-26q3u-zimva-as2tg-dae",
        "sub_account":{
            "address": "0000000000000000000000000000000000000000000000000000000000000000"
        }
       }
}'
```

> The response lists all transactions where the given `account_identifier` was involved in:

Reviewer: ```suggestion
The response lists all transactions where the given `account_identifier` was involved:
```
